partitionmanager/bootstrap.py

```diff
@@ -101,6 +117,91 @@ def _plan_partitions_for_time_offsets(
     return changes
 
 
+def _suffix(lines, *, indent="", mid_suffix="", final_suffix=""):
+    """ Helper that suffixes each line with either mid- or final- suffix """
+    for line, is_final in partitionmanager.tools.iter_show_end(lines):
+        if is_final:
+            yield indent + line + final_suffix
+        else:
+            yield indent + line + mid_suffix
+
+
+def _trigger_column_copies(cols):
+    """ Helper that returns lines copying each column for a trigger. """
+    for c in cols:
+        yield f"`{c}` = NEW.`{c}`"
+
+
+def _generate_sql_copy_commands(
+    existing_table, map_data, columns, new_table, alter_commands_iter
+):
+    """ Generate a series of SQL commands to start a copy of the existing_table
+    to a new_table, applying the supplied alterations before starting the
+    triggers. """
+    log = logging.getLogger(
+        f"_generate_sql_copy_commands:{existing_table.name} to {new_table.name}"
+    )
+
+    max_val_part = map_data["partitions"][-1]
+    if not isinstance(max_val_part, partitionmanager.types.MaxValuePartition):
+        log.error(f"Expected a MaxValue partition, got {max_val_part}")
+        raise Exception("Unexpected part?")
+
+    range_id_string = ", ".join(map_data["range_cols"])
+
+    if len(map_data["range_cols"]) == 1:
+        range_cols_string = "RANGE"
+        max_val_string = "MAXVALUE"
+    else:
+        num_cols = len(map_data["range_cols"])
+        range_cols_string = "RANGE COLUMNS"
+        max_val_string = "(" + ", ".join(["MAXVALUE"] * num_cols) + ")"
```

Comment:
nit: This duplicates a bit of the logic in `MaxValuePartition.values` but I'm not sure if it'd be worth the effort of instantiating one of those here.

Reply:
Yeah, seems like the reuse value is pretty low, while that would increase a fair amount of coupling or further obfuscate the MaxValuePartition type